File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/payload/TicketValidityResponse.java
package de.tum.in.tca.ticketcheck.component.ticket.payload;

import com.google.gson.annotations.SerializedName;

public class TicketValidityResponse {

    // TODO: refactor this to Kotlin class!

    @SerializedName("valid")
    public boolean valid;

    @SerializedName("ticket_history")
    public int ticketHistory;

    @SerializedName("firstname")
    public String firstName;

    @SerializedName("name")
    public String lastName;

    @SerializedName("tumid")
    public String tumID;

    @SerializedName("purchasedate")
    public String purchaseDate;

    @SerializedName("redeemdate")
    public String redeemDate;

    @SerializedName("status")
    public String status;

    public String getTicketInfo() {
        return firstName + " " + lastName + "\n" + tumID + "\n" + purchaseDate + "\n" + redeemDate + "\n" + status;

    }
}

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/payload/TicketValidityResponse.kt
package de.tum.`in`.tca.ticketcheck.component.ticket.payload

import com.google.gson.annotations.SerializedName
import de.tum.`in`.tca.ticketcheck.component.ticket.model.TicketWithRedemption

data class TicketValidityResponse(
        @SerializedName("valid") var valid: Boolean = false,
        @SerializedName("tumid") var tumID: String? = null,
        @SerializedName("name") var lastName: String? = null,
        var tickets: List<TicketWithRedemption> = emptyList()
)
